wikidataintegrator/wdi_helpers/publication.py

```diff
@@ -192,20 +198,22 @@ def validate(self):
     def make_reference(self):
         if self.source == "crossref":
             assert 'doi' in self.ids
-            extid = wdi_core.WDString(self.ids['doi'], PROPS['DOI'], is_reference=True)
-            ref_url = wdi_core.WDUrl(self.ref_url, PROPS['reference URL'], is_reference=True)
-            stated_in = wdi_core.WDItemID(self.SOURCES[self.source], PROPS['stated in'], is_reference=True)
-            retrieved = wdi_core.WDTime(strftime("+%Y-%m-%dT00:00:00Z", gmtime()), PROPS['retrieved'],
-                                        is_reference=True)
-            self.reference = [stated_in, extid, ref_url, retrieved]
+            edt_id_id, ext_id_prop = self.ids['doi'], PROPS['DOI']
         elif self.source == "europepmc":
             assert 'pmcid' in self.ids
-            extid = wdi_core.WDString(self.ids['pmcid'], PROPS['PMCID'], is_reference=True)
-            ref_url = wdi_core.WDUrl(self.ref_url, PROPS['reference URL'], is_reference=True)
-            stated_in = wdi_core.WDItemID(self.SOURCES[self.source], PROPS['stated in'], is_reference=True)
-            retrieved = wdi_core.WDTime(strftime("+%Y-%m-%dT00:00:00Z", gmtime()), PROPS['retrieved'],
-                                        is_reference=True)
-            self.reference = [stated_in, extid, ref_url, retrieved]
+            edt_id_id, ext_id_prop = self.ids['pmcid'], PROPS['PMCID']
+        elif self.source == 'arxiv':
+            assert 'arxiv' in self.ids
+            edt_id_id, ext_id_prop = self.ids['arxiv'], PROPS['arxiv id']
+        else:
+            raise ValueError(f'Unhandled source: {self.source}')
```

Comment:
I also added a raise in case an invalid source is used, because when I was updating the code, I didn't know I needed to change something here and was really surprised by the `None` that was being returned from this function